component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateAuthenticator.java

```diff
@@ -604,7 +604,11 @@ private String getUserStoreDomainName(String userIdentifier, AuthenticationConte
                 throw new AuthenticationFailedException("Unable to find X509 Certificate's user in user store. ");
             }
         } else {
-            return UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME;
+            if (userIdentifier.indexOf("/") > 0) {
```

Comment:
if the property is not enabled should we consider other user store?Previous behavior was if the property is enabled only we can use domain separator in the username right?

Reply:
in previous case if the property is enabled always return as 'PRIMARY'. let's take certificate which has CN STORE1/user1. Then even though property is not enabled isUserExists returns true since domain name is given (userName will be fully qualified name). In that case if we return 'PRIMARY' from the function getUserStoreDomainName it will be wrong right?